### ISSUE TITLE
Nose with Python 3.10

### ISSUE THREAD

**Issue by skiedude** [StackStorm-Exchange/ci#154]
While looking into the CI failures on linked packs on https://github.com/StackStorm-Exchange/ci/issues/153 it appears the python package `nose` is causing a lot of failures.

While I didn't check all linked packs, of the 4 I checked 2 were failing due to `nose` on Python 3.10. But I assume every CI testing in Python3.10 using this CI as a base is going to fail for this issue.

https://github.com/StackStorm-Exchange/stackstorm-github/actions/runs/12326010817/job/34406161027#step:6:378
https://github.com/StackStorm-Exchange/stackstorm-terraform/actions/runs/12326013523/job/34406167325#step:6:338

`nose` uses `collections.Callable` but in python 3.10 that was moved to `collections.abc.Callable`  which is causing an attribute error `AttributeError: module 'collections' has no attribute 'Callable'` https://github.com/nose-devs/nose/issues/1122 . Unfortunately `nose` is also abandoned, so no official fix in this module is going to take place.

`pynose` https://pypi.org/project/pynose/ looks to be a currently maintained successor. As i'm not familiar with `nose` at all, I'm not sure if its going to be an easy straight across replacement of package. 

However I think a good chunk of the failing CI tests are atleast initially from this issue.

**Comment by skiedude** [comment]
I can submit a PR for https://github.com/StackStorm-Exchange/ci/blob/master/.github/actions/py-dependencies/requirements-pack-tests.txt to use `pynose` and see how the CI in this repo handles it?

**Comment by skiedude** [comment]
Ok I decided just to try and get it to work with my own fork of the ci repo.

Using this ci fork https://github.com/skiedude/ci/tree/python I had to make a handful of changes to get `pynose` installed. 

I found the following:
1. all the packages in `ci/blob/master/.github/actions/py-dependencies/requirements-pack-tests.txt` are being ignored because they don't pin a version `==`
2. I added pynose with a version, which caused it to be installed!
3. However `nose` was still being installed. This was a result of the main https://github.com/StackStorm/st2/blob/master/requirements.txt#L36 actually being the location where nose was installed in the CI. 
4. I added a custom step to uninstall nose and force reinstall pynose to get around this temporarily. 

With all of the above changes, the CI on stackstorm-github passes on 3.10 (as well as 3.8, 3.9)
https://github.com/StackStorm-Exchange/stackstorm-github/pull/57
